### PR TITLE
Scatter Graph - Update to support the allowXCallibration prop

### PR DESCRIPTION
**Pull request by artpark**
[UXPLATFORM-6842](https://jira2.cerner.com/browse/UXPLATFORM-6824)
===

### Summary
Allows a scatter graph to be configured with the 'Allow Callibration' preference. This allows the user to see all the data points without it being cut off.

### Deployment Link
https://terra-graphs-deployed-pr-#.herokuapp.com/

### Testing
`npm run karma`
Screen Shot 2022-08-31 at 11 47 35 AM

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/carbon-graphs/src/js/controls/Scatter/Scatter.js [hunk 1]
```diff
@@ -45,16 +45,43 @@ import { validateData } from '../../helpers/constructUtils';
...
+ */
+/**
+ * Calculates the min and max values for X-axis.
+ * First we filter out values that are `null`, this is a result of
+ * datapoint being part of being in a non-contiguous series and then we
+ * get the min and max values for the X-axis domain.
+ *
+ * @private
+ * @param {Array} values - Datapoint values
+ * @returns {object} - Contains min and max values for the data points for Y and Y2 axis
```
Comment (sdadn) [inline packages/carbon-graphs/src/js/controls/Scatter/Scatter.js:71]:
```suggestion
 * @returns {object} - Contains min and max values for the data points for the x-axis
```

File: packages/carbon-graphs/src/js/controls/Scatter/Scatter.js [hunk 2]
```diff
@@ -233,10 +266,7 @@ class Scatter extends GraphContent {
       .remove();
 
     updateShapesDuringReflow(graph, graphData, this);
-    this.valuesRange = calculateValuesRange(
-      this.config.values,
-      this.config.yAxis,
-    );
+    this.valuesRange[this.config.yAxis] = calculateValuesRangeYAxis(this.config.values, this.config.yAxis);
```
Comment (sdadn) [inline packages/carbon-graphs/src/js/controls/Scatter/Scatter.js:269]: The second parameter is no longer used (I missed this in the previous PRs, will correct them soon).
```suggestion
    this.valuesRange[this.config.yAxis] = calculateValuesRangeYAxis(this.config.values);
```

Comment (artpark) [reply packages/carbon-graphs/src/js/controls/Scatter/Scatter.js]: Removed 49c8fce

File: packages/carbon-graphs/src/js/controls/Scatter/Scatter.js (hunk 1 above)
Comment (artpark) [reply packages/carbon-graphs/src/js/controls/Scatter/Scatter.js]: Fixed 49c8fce

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js [hunk 3]
```diff
@@ -296,6 +296,100 @@ describe('Scatter - Load', () => {
         input.key,
       );
     });
+    it('does not update x axis range if allow calibration is disabled', () => {
+      const disableCalibrationInput = getAxes(axisDefault);
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js:300]: I would rename `disableCalibrationInput` to `graphConfig` and
`disableCalibrationGraph` to `graphInstance` like I did here: https://github.com/cerner/terra-graphs/pull/288/files

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js [hunk 4]
```diff
@@ -296,6 +296,100 @@ describe('Scatter - Load', () => {
...
+      disableCalibrationInput.axis.x.allowCalibration = false;
+      disableCalibrationInput.axis.x.lowerLimit = 50;
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js:302]: Is there a reason why only `lowerLimit` is being set here?

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js (hunk 3 above)
Comment (artpark) [reply packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js]: Renamed 25980a5

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js (hunk 4 above)
Comment (artpark) [reply packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js]: No reason, it has been removed 25980a5

**Review by sdadn — APPROVED** [review]

**Review by benbcai — APPROVED** [review]

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js [hunk 5]
```diff
@@ -296,6 +296,98 @@ describe('Scatter - Load', () => {
...
+      const graphInstance = new Graph(graphConfig);
+      input = getInput(valuesDefault, false, false);
+      graphInstance.loadContent(new Scatter(input));
+
+      expect(graphInstance.config.axis.x.domain.upperLimit)
+        .toEqual(graphConfig.axis.x.upperLimit);
+      expect(graphInstance.config.axis.x.domain.lowerLimit)
+        .toEqual(graphConfig.axis.x.lowerLimit);
+    });
+    it('does not x axis range if allowCalibration is true and datapoints are within limits', () => {
```
Comment (sycombs) [inline packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js:323]: nit
```suggestion
    it('does not update x axis range if allowCalibration is true and datapoints are within limits', () => {
```

**Review by sycombs — APPROVED** [review]

File: packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js (hunk 5 above)
Comment (artpark) [reply packages/carbon-graphs/tests/unit/controls/Scatter/ScatterLoad-spec.js]: Fixed 5fb6adc

**Pull request merged**